### PR TITLE
[Merged by Bors] - refactor(RingTheory/Ideal/Pointwise): Generalize to semirings

### PR DESCRIPTION
**Pull request by tb65536**
This PR generalizes much of `RingTheory/Ideal/Pointwise` to semirings. I also moved a couple lemmas from the `Group` section to the `Monoid` section, and added a lemma to `RingTheory/Ideal/Over`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Ideal/Pointwise.lean [hunk 1]
```diff
@@ -28,26 +28,34 @@ namespace Ideal
...
+open Pointwise
+
+/-- The action on an ideal corresponding to applying the action to every element.
+
+This is available as an instance in the `Pointwise` locale. -/
+protected def pointwiseMulSemiringAction {R : Type*} [CommRing R] [MulSemiringAction M R] :
+    MulSemiringAction M (Ideal R) where
   smul_one a := by simp only [Ideal.one_eq_top]; exact Ideal.map_top _
   smul_mul a I J := Ideal.map_mul (MulSemiringAction.toRingHom _ _ a) I J
   smul_add _ I J := Ideal.map_sup _ I J
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Pointwise.lean:54]: This holds in the non-commutative case, right? Can you add a `DistribMulAction M (Ideal R)` instance as well?

Comment (eric-wieser) [reply Mathlib/RingTheory/Ideal/Pointwise.lean:54]: Ah nevermind, I think you can just change the `MulAction` above to a `DistribMulAction`

Comment (tb65536) [reply Mathlib/RingTheory/Ideal/Pointwise.lean]: Thanks!

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**